nuxeo-features/nuxeo-automation/nuxeo-automation-core/src/test/java/org/nuxeo/ecm/automation/core/operations/document/CopyDocumentTest.java

```diff
@@ -0,0 +1,128 @@
+ * limitations under the License.
+ *
+ * Contributors:
+ *     Nour AL KOTOB
+ */
+package org.nuxeo.ecm.automation.core.operations.document;
+
+import static org.junit.Assert.assertEquals;
+
+import java.io.Serializable;
+import java.util.Map;
+
+import org.junit.After;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.nuxeo.ecm.automation.AutomationService;
+import org.nuxeo.ecm.automation.OperationContext;
+import org.nuxeo.ecm.automation.OperationException;
+import org.nuxeo.ecm.core.api.CoreSession;
+import org.nuxeo.ecm.core.api.DocumentModel;
+import org.nuxeo.ecm.core.api.event.CoreEventConstants;
+import org.nuxeo.ecm.core.test.CoreFeature;
+import org.nuxeo.runtime.test.runner.Deploy;
+import org.nuxeo.runtime.test.runner.Features;
+import org.nuxeo.runtime.test.runner.FeaturesRunner;
+import org.nuxeo.runtime.test.runner.TransactionalFeature;
+
+import javax.inject.Inject;
+
+/**
+ * @since 11.1
+ */
+@RunWith(FeaturesRunner.class)
+@Features(CoreFeature.class)
+@Deploy("org.nuxeo.ecm.automation.core")
+public class CopyDocumentTest {
+    public static final String ROOT = "/";
+
+    public static final String COPY_DOC_NAME = "copyDoc";
+
+    public static final String TARGET_PROPERTY_KEY = "target";
+
+    public static final String NAME_PROPERTY_KEY = "name";
+
+    public static final String FILE = "File";
+
+    @Inject
+    protected CoreSession session;
+
+    @Inject
+    protected AutomationService automationService;
+
+    @Inject
+    protected TransactionalFeature txFeature;
+
+    @After
+    public void tearDown() {
+        session.removeChildren(session.getRootDocument().getRef());
+        session.save();
+    }
+
+    @Test
+    public void testCopyChildren() throws OperationException {
+        DocumentModel folder = session.createDocumentModel(ROOT, "SourceFolder", "Folder");
+        folder = session.createDocument(folder);
+
+        DocumentModel childDoc = session.createDocumentModel(folder.getPathAsString(), "childDoc", FILE);
+        session.createDocument(childDoc);
+
+        OperationContext context = new OperationContext(session);
+        context.setInput(folder);
+
+        Map<String, Serializable> params = Map.of(TARGET_PROPERTY_KEY, ROOT, NAME_PROPERTY_KEY, "CopyFolder");
+        DocumentModel result = (DocumentModel) automationService.run(context, CopyDocument.ID, params);
+
+        result = session.getDocument(result.getRef());
+        assertEquals(1, session.getChildren(result.getRef()).size());
+    }
+
+    @Test
+    public void testLifeCycleResetOnCopy() throws OperationException {
+        DocumentModel source = session.createDocumentModel(ROOT, "Source", FILE);
+        source = session.createDocument(source);
+
+        String initialLifeCycleState = source.getCurrentLifeCycleState();
+
+        // Change the source current lifecycle state
+        source.followTransition("approve");
+        source = session.saveDocument(source);
```

Comment:
`followTransition` already writes the document model, furthermore follow transition implementation never touch the given document model, see:
https://github.com/nuxeo/nuxeo/blob/9db13fe5cddbf08dc5be932a52a7501f70556e5c/nuxeo-core/nuxeo-core/src/main/java/org/nuxeo/ecm/core/api/AbstractSession.java#L2065

`saveDocument` shouldn't be needed.